lib/models/rating_model.dart

```diff
@@ -0,0 +1,18 @@
+class RatingModel {
+  final String id;
+  final double rating;
+
+  RatingModel({this.id, this.rating});
+
+  Map<String, dynamic> toJson() => {
+        'id': id,
+        'rating': rating,
+      };
+
+  factory RatingModel.fromJSON(Map<String, dynamic> json) {
+    return RatingModel(
+      id: json['id'],
+      rating: json['rating'],
+    );
+  }
```

Comment:
The response body that you receive from PUT operation on user's table will be an updated user instance, not id and rating. This is what gives you an error `type 'List<dynamic>' is not a subtype of type 'Map<String, dynamic>'`. Since response body is a user instance, you can remove this part and use UserModel's `fromJson` function in the rating service file.